umpost/um2netcdf.py

```diff
@@ -736,6 +733,34 @@ def fix_level_coord(cube, z_rho, z_theta, tol=1e-6):
                 c_sigma.var_name = 'sigma_theta'
 
 
+MAX_NP_INT32 = np.iinfo(np.int32).max
+MIN_NP_INT32 = np.iinfo(np.int32).min
+
+
+def convert_32_bit(cube):
+    """
+    Convert 64 bit int/float data to 32 bit (in place).
+
+    Parameters
+    ----------
+    cube : iris.cube object to modify.
+    """
+    if cube.data.dtype == 'float64':
+        cube.data = cube.data.astype(np.float32)
+    elif cube.data.dtype == 'int64':
+        _max = np.max(cube.data)
+        _min = np.min(cube.data)
+
+        if _max > MAX_NP_INT32:
+            msg = f"Converting {cube.var_name} causes a 32 bit overflow!"
+            warnings.warn(msg)
```

Comment:
It would be best to explicitly define which warning class to return. The default `UserWarning` is probably OK, although `RuntimeWarning` may be more correct.

Reply:
The thought occurred that `numpy` might include an option to toggle overflow warnings. This is based on seeing `numpy 2.x` raise emit more warnings than some code run against `v1.22`. I'll check the docs it'd be great to reuse existing warnings.

Reply:
A quick search shows auto detecting & warning of integer overflow in arrays is still an open issue:
https://github.com/numpy/numpy/issues/8987

I'll treat this as a case requiring manual checks & emitting a relevant warning. Performance is a possible issue, although I'm deliberately just checking the max value for some efficiency (I'm unsure if `numpy` caches min/max values). Some other ways around this are:
* review min/max values for each data variable
  - Are any vars outside `-/+ 2**31` (apply overflow checks on selected vars)
  - Detect large or odd values during data reads/sanity checking process?
  - Could be a lot of work...
* deliberately skip warnings via a new option

Reply:
I think checking the min/max for now is satisfactory. It would also be good to have the ability to turn off the warnings if you're certain you don't need them/you don't care.

Reply:
> It would be best to explicitly define which warning class to return. The default `UserWarning` is probably OK, although `RuntimeWarning` may be more correct.

According to https://docs.python.org/3.10/library/exceptions.html#RuntimeWarning, the Runtime one is a _Base class for warnings about dubious runtime behavior_. That's pretty apt to me :-)

Reply:
> It would be best to explicitly define which warning class to return. The default `UserWarning` is probably OK, although `RuntimeWarning` may be more correct.

I've fixed this in 15274011 to use `RuntimeWarning`.

Reply:
> I think checking the min/max for now is satisfactory. It would also be good to have the ability to turn off the warnings if you're certain you don't need them/you don't care.

I've expanded #16 to cover this/fixing command line issues as a whole.